api/test/logs/logger_benchmark.cc

```diff
@@ -0,0 +1,260 @@
+
+    MultiThreadRunner(state, [&logger]() {
+      for (int64_t i = 0; i < kMaxIterations; i++)
+      {
+        logger->Trace(
+            "This is a simple structured log message from {process_id}:{thread_id}",
+            opentelemetry::common::MakeAttributes({{"process_id", 12347}, {"thread_id", 12348}}));
+      }
+    });
+
+    state.ResumeTiming();
+  }
+}
+BENCHMARK(BM_StructuredLogWithTwoAttributes);
+
+static void BM_StructuredLogWithEventIdAndTwoAttributes(benchmark::State &state)
+{
+  auto lp     = Provider::GetLoggerProvider();
+  auto logger = lp->GetLogger("StructuredLogWithEventId");
+
+  for (auto _ : state)
+  {
+    state.PauseTiming();
+
+    MultiThreadRunner(state, [&logger]() {
+      for (int64_t i = 0; i < kMaxIterations; i++)
+      {
+        logger->Trace(
+            "This is a simple structured log message from {process_id}:{thread_id}",
+            opentelemetry::common::MakeAttributes({{"process_id", 12347}, {"thread_id", 12348}}));
+      }
+    });
+
+    state.ResumeTiming();
+  }
+}
+BENCHMARK(BM_StructuredLogWithEventIdAndTwoAttributes);
+
+static void BM_StructuredLogWithEventIdStructAndTwoAttributes(benchmark::State &state)
+{
+  auto lp     = Provider::GetLoggerProvider();
+  auto logger = lp->GetLogger("StructuredLogWithEventId");
+
+  const EventId function_name_event_id{0x12345678, "Company.Component.SubComponent.FunctionName"};
+
+  for (auto _ : state)
+  {
+    state.PauseTiming();
+
+    MultiThreadRunner(state, [&logger, &function_name_event_id]() {
+      for (int64_t i = 0; i < kMaxIterations; i++)
+      {
+        logger->Trace(
+            function_name_event_id,
+            "Simulate function enter trace message from {process_id}:{thread_id}",
+            opentelemetry::common::MakeAttributes({{"process_id", 12347}, {"thread_id", 12348}}));
+      }
+    });
+
+    state.ResumeTiming();
+  }
+}
+BENCHMARK(BM_StructuredLogWithEventIdStructAndTwoAttributes);
+
+static void BM_EnabledOnSeverityReturnFalse(benchmark::State &state)
+{
+  auto lp     = Provider::GetLoggerProvider();
+  auto logger = lp->GetLogger("UnstructuredLogWithEnabledReturnFalse");
+
+  for (auto _ : state)
+  {
+    state.PauseTiming();
+
+    MultiThreadRunner(state, [&logger]() {
+      for (int64_t i = 0; i < kMaxIterations; i++)
+      {
+        if (logger->Enabled(Severity::kTrace))
+        {
+          logger->Trace("This log line would not be called");
+        }
+      }
+    });
+
+    state.ResumeTiming();
+  }
+}
+BENCHMARK(BM_EnabledOnSeverityReturnFalse);
+
+static void BM_EnabledOnSeverityAndEventIdReturnFalse(benchmark::State &state)
+{
+  auto lp     = Provider::GetLoggerProvider();
+  auto logger = lp->GetLogger("UnstructuredLogWithEnabledReturnFalse");
+
+  for (auto _ : state)
+  {
+    state.PauseTiming();
+
+    MultiThreadRunner(state, [&logger]() {
+      for (int64_t i = 0; i < kMaxIterations; i++)
+      {
+        if (logger->Enabled(Severity::kTrace, 0x12345678), false)
```

Comment:
Does this really build ? How ?

Reply:
Thanks for the catch. It builds but the trailing `, false` should be removed. The variation comes from an iteration of experiment change, but I just validated that the benchmark result is consistent after removing the extra experiment code.